src/Oro/Bundle/CacheBundle/DependencyInjection/Compiler/CacheWarmerPass.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Oro\Bundle\CacheBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+use Oro\Bundle\CacheBundle\Provider\ConfigCacheWarmerInterface;
+use Oro\Component\Config\Dumper\ConfigMetadataDumperInterface;
+
+class CacheWarmerPass implements CompilerPassInterface
+{
+    const SERVICE_ID = 'oro_cache.config_cache_warmer.listener';
+    const PROVIDER_TAG = 'oro.config_cache_warmer.provider';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition(self::SERVICE_ID)) {
+            return;
+        }
+
+        $service = $container->getDefinition(self::SERVICE_ID);
+        $cacheProviders = $container->findTaggedServiceIds(self::PROVIDER_TAG);
+
+        foreach ($cacheProviders as $id => $attributes) {
+            $definition = $container->getDefinition($id);
+            $this->assertInstanceOf($definition, $container, ConfigCacheWarmerInterface::class);
+
+            if (isset($attributes[0]['dumper'])) {
+                $dumper = $attributes[0]['dumper'];
+            } else {
+                throw new \InvalidArgumentException(
+                    sprintf('Attribute "dumper" is required for config cache warmer %s', $id)
+                );
+            }
+
+            $this->assertDumper($container, $dumper);
+
+            $service->addMethodCall(
+                'addConfigCacheProvider',
+                [new Reference($id), new Reference($dumper)]
+            );
+        }
+    }
+
+    /**
+     * @param Definition $definition
+     * @param ContainerBuilder $container
+     * @param string $class
+     */
+    protected function assertInstanceOf(Definition $definition, ContainerBuilder $container, $class)
+    {
+        $className = $definition->getClass();
+
+        if (!class_exists($className)) {
```

Comment:
I guess it can be safely replaced with
```
        if (0 === strpos($className, '%')) {
            $className = $container->getParameter(substr($className, 1, -1));
        }
```
what do you think?